test/functional/core/path_spec.lua

```diff
@@ -103,6 +103,14 @@ describe('file search', function()
     eq('filename_with_unicode_ααα', eval('expand("%:t")'))
   end)
 
+  it('find file name with prefix `file://` in line #24032', function()
+    command('cd test/functional/fixtures')
+    insert('file://filename_with_unicode_ααα')
+    eq('', eval('expand("%")'))
+    feed('gf')
+    eq('filename_with_unicode_ααα', eval('expand("%:t")'))
```

Comment:
need tests for `file:///` also. and other edge cases like `https://example.com/file:/foo.txt` . 

see also https://github.com/microsoft/vscode-uri/blob/53e4ca6263f2e4ddc35f5360c62bc1b1d30f27dd/src/test/uri.test.ts#L264-L268

Reply:
Added the test and not handling `file://` (also added test cases for each of them)

PTAL